apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx

```diff
@@ -136,23 +136,22 @@ export const DailySchedule = ({
     defaultSelectedService
   );
 
+  const getJourneysForSelectedService = (service: Service): void => {
+    fetch({
+      fetcher: fetchJourneys(routeId, stopId, service, directionId, false),
+      parser: parseResults
+    });
+  };
+
   useEffect(
     () => {
       /* istanbul ignore next */
       if (selectedService && isNotStarted(fetchState)) {
```

Comment:
```suggestion
      if (selectedService && !isLoading(fetchState)) {
```

Would changing `isNotStarted` to `!isLoading` allow it to re-fetch properly?

Reply:
I tried that; that change causes an infinite loop because once the data is retrieved, status changes to "with data", which evaluates to true for `isLoading(fetchState)` and hence fetching again.

Reply:
Ah, ok, thanks. That makes me wonder if `fetchState` shouldn't have been in that dependency list. But what you've done looks great, this is all just idle curiosity on my part.